### PR TITLE
fix: improve original author / translator feature

### PR DESCRIPTION
**Pull request by scissorsneedfoodtoo**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Closes #XXXXX

This PR adds the original author / translator feature for News editors, and should help simplify the process of attributing the original author for any translations.

To use this feature, editors will need to add a `script` tag to either the post head or post foot of the article from the Ghost dashboard under "Code injection". The script tag should contain a variable named `fCCOriginalPost` set equal to the full URL of the original post.

For example, if someone translate's one of Quincy's latest articles into another language, they can add the following code to the post head or post footer:

```js
<script>
  const fCCOriginalPost = 'https://www.freecodecamp.org/news/python-split-string/';
</script>
```

Then during the build, info about the original article will be appended to the translated post, and details about the original author will be added to the post's card on the landing page, author page, and tag page. Details about the original article and translator will also be added to the top and bottom bylines of the translated post, and some intro text will be added to the beginning of the article.

Below are some screenshots of the new feature taken from a Spanish Dockerized Ghost build.

Post card on the landing page (should also look the same on the tag and Rafael's profile page):

image

Byline at the top of the post:

image

Intro text:

image

Byline with bio at the bottom of the post:

image

Also, the tweet can change depending on whether both the original author and translator have a Twitter account. Here's what the message looks like if both original author and translator are on Twitter:

image

-----

One of the bigger changes, besides those to the templates and translation schema, have to do with the way we handle all the different API instances.

Now the process should be flexible enough to handle changes to the Ghost URL, like the move of Chinese from a subdomain to a subpath. I believe it should also be able to handle the Ghost dashboards moving to a subdomain like `write`, though we'll want to do some testing.

Also, this feature should be able to handle translations from any active language into another. For example, English to Japanese, or Italian to Portuguese.

The only catch is that this feature won't work with Chinese to X language until Chinese is moved to a subpath. Because both the English and Chinese paths look similar, we'll fall back to English for the time being.

It will take some time to test this with each locale, so I'm thinking it might be best to review this as-is. If it's merged, editors can try out the feature with a few articles and test the different views, the Twitter button, and so on. Then we can fix bugs as they're reported.

Also, the search indexer will need to be updated to support this feature on the search results page. A mock is there for testing, but it will not work until the indexer is updated along with the indexed records.

### CODE REVIEW

File: i18n/locales/italian/translations.json [hunk 1]
```diff
@@ -32,23 +32,35 @@
...
-    "author-has-twitter": "Se hai letto fino a qui, twitta al traduttore per ringraziarlo. {{ <0> }}Twitta un ringraziamento{{ </0> }}",
-    "author-no-twitter": "Se questo articolo ti è stato utile, {{ <0> }}twittalo{{ </0> }}.",
-    "default-tweet": "Grazie {{ twitter }} per avere tradotto questo utile articolo."
+    "cta": {
+      "tweet-a-thanks": "Se hai letto fino a qui, twitta al traduttore per ringraziarlo. {{ <0> }}Twitta un ringraziamento{{ </0> }}",
+      "tweet-it": "Se questo articolo ti è stato utile, {{ <0> }}twittalo{{ </0> }}."
+    },
+    "tweets": {
+      "default": "Grazie {{ author }} per avere tradotto questo utile articolo.",
+      "translation": "Grazie {{ author }} per avere tradotto questo utile articolo e {{ translator }} per averlo tradotto."
```
Comment (majestic-owl448) [inline i18n/locales/italian/translations.json]:
```suggestion
      "translation": "Grazie {{ author }} per avere scritto questo utile articolo e {{ translator }} per averlo tradotto."
```

File: i18n/locales/italian/translations.json [hunk 2]
```diff
@@ -32,23 +32,35 @@
...
-      "pt-br": "Portoghese",
-      "it": "Italiano"
+    "locales": {
+      "english": "inglese",
+      "espanol": "spagnolo",
+      "chinese": "cinese",
+      "portuguese": "portoghese",
+      "italian": "italiano",
+      "japanese": "giapponese",
+      "arabic": "araba",
```
Comment (majestic-owl448) [inline i18n/locales/italian/translations.json]:
```suggestion
      "arabic": "arabo",
```

File: i18n/locales/italian/translations.json [hunk 3]
```diff
@@ -32,23 +32,35 @@
...
   "social-row": {
-    "author-has-twitter": "Se hai letto fino a qui, twitta al traduttore per ringraziarlo. {{ <0> }}Twitta un ringraziamento{{ </0> }}",
-    "author-no-twitter": "Se questo articolo ti è stato utile, {{ <0> }}twittalo{{ </0> }}.",
-    "default-tweet": "Grazie {{ twitter }} per avere tradotto questo utile articolo."
+    "cta": {
+      "tweet-a-thanks": "Se hai letto fino a qui, twitta al traduttore per ringraziarlo. {{ <0> }}Twitta un ringraziamento{{ </0> }}",
+      "tweet-it": "Se questo articolo ti è stato utile, {{ <0> }}twittalo{{ </0> }}."
+    },
+    "tweets": {
+      "default": "Grazie {{ author }} per avere tradotto questo utile articolo.",
```
Comment (majestic-owl448) [inline i18n/locales/italian/translations.json]:
```suggestion
      "default": "Grazie {{ author }} per avere scritto questo utile articolo.",
```

File: i18n/locales/italian/translations.json [hunk 4]
```diff
@@ -32,23 +32,35 @@
     "go-to-front-page": "Vai alla Home"
   },
   "social-row": {
-    "author-has-twitter": "Se hai letto fino a qui, twitta al traduttore per ringraziarlo. {{ <0> }}Twitta un ringraziamento{{ </0> }}",
-    "author-no-twitter": "Se questo articolo ti è stato utile, {{ <0> }}twittalo{{ </0> }}.",
-    "default-tweet": "Grazie {{ twitter }} per avere tradotto questo utile articolo."
+    "cta": {
+      "tweet-a-thanks": "Se hai letto fino a qui, twitta al traduttore per ringraziarlo. {{ <0> }}Twitta un ringraziamento{{ </0> }}",
```
Comment (majestic-owl448) [inline i18n/locales/italian/translations.json]:
```suggestion
      "tweet-a-thanks": "Se hai letto fino a qui, {{ <0> }}scrivi un tweet di ringraziamento!"{{ </0> }}",
```

File: i18n/locales/chinese/translations.json [hunk 5]
```diff
@@ -24,7 +24,7 @@
     "no-posts": "没有文章",
     "one-post": "1 篇文章",
     "multiple-posts": "{{ postCount }} 篇文章",
-    "read-more": "阅读该{{ role }}的{{ <0> }}更多文章{{ </0> }}。"
+    "read-more": "阅读的{{ <0> }}更多文章{{ </0> }}。"
```
Comment (miyaliu666) [inline i18n/locales/chinese/translations.json:27]: 阅读{{ role }}的{{ <0> }}更多文章{{ </0> }}。

File: i18n/locales/chinese/translations.json [hunk 6]
```diff
@@ -24,7 +24,7 @@
...
   },
   "tag": {
     "no-posts": "文章汇总",
```
Comment (miyaliu666) [inline i18n/locales/chinese/translations.json]: "no-posts": "没有文章",

（The original translation means list of articles - I guess it's a mistake I made before?）

File: i18n/locales/chinese/translations.json [hunk 7]
```diff
@@ -36,18 +36,25 @@
...
+      "translated-by": "{{ <0> }}译者：{{ </0> }} {{ translator }}"
+    },
+    "locales": {
+      "english": "英语",
+      "espanol": "西班牙语",
+      "chinese": "中文",
+      "portuguese": "葡萄牙语",
+      "italian": "意大利语",
+      "japanese": "日语",
+      "arabic": "阿拉伯",
```
Comment (miyaliu666) [inline i18n/locales/chinese/translations.json]: 阿拉伯语

File: i18n/locales/chinese/translations.json [hunk 8]
```diff
@@ -36,18 +36,25 @@
...
+      "urdu": "乌尔都语",
+      "bengali": "孟加拉"
```
Comment (miyaliu666) [inline i18n/locales/chinese/translations.json]: 孟加拉语

File: i18n/locales/chinese/translations.json (hunk 5 above)
Comment (majestic-owl448) [inline i18n/locales/chinese/translations.json]:
```suggestion
    "read-more": "阅读{{ role }}的{{ <0> }}更多文章{{ </0> }}。"
```

Comment (majestic-owl448) [reply i18n/locales/chinese/translations.json]: @miyaliu666 I don't think there is a `{{role}}` variable anymore tho

File: i18n/locales/espanol/translations.json [hunk 9]
```diff
@@ -32,23 +32,35 @@
...
-    "author-has-twitter": "Si leíste hasta aquí, haz un tweet al autor para mostrarle que te importa su trabajo. {{ <0> }}Realiza un Tweet de agradecimiento{{ </0> }}",
-    "author-no-twitter": "Si este artículo te fue útil, {{ <0> }}haz un tweet{{ </0> }}.",
-    "default-tweet": "Gracias {{ twitter }} por escribir este útil artículo."
+    "cta": {
+      "tweet-a-thanks": "Si leíste hasta aquí, haz un tweet al autor para mostrarle que te importa su trabajo. {{ <0> }}Realiza un Tweet de agradecimiento{{ </0> }}",
+      "tweet-it": "Si este artículo te fue útil, {{ <0> }}haz un tweet{{ </0> }}."
+    },
+    "tweets": {
+      "default": "Gracias {{ author }} por escribir este útil artículo.",
+      "translation": "Gracias {{ author }} por escribir este útil artículo y {{ translator }} por traducirlo y adaptado."
```
Comment (RafaelDavisH) [inline i18n/locales/espanol/translations.json]: "translation": "Gracias {{ author }} por escribir este útil artículo y {{ translator }} por tu traducción."

**Review by RafaelDavisH — COMMENTED** [review]
Left a comment on line 41

**Review by RafaelDavisH — CHANGES_REQUESTED** [review]
"translation": "Gracias {{ author }} por escribir este útil artículo y {{ translator }} por tu traducción."

File: i18n/locales/urdu/translations.json [hunk 10]
```diff
@@ -32,23 +32,35 @@
...
+  "original-author-translator": {
     "roles": {
-      "author": "مصنف",
-      "translator": "مترجم"
+      "author": "مصنف: {{ name }} ({{ locale }})",
+      "translator": "مترجم: {{ name }}"
+    },
+    "details": {
+      "original-article": "{{ <0> }}اصل مضمون:{{ </0> }} {{ title }} از {{ author }}",
+      "translated-by": "{{ <0> }}ترجمہ از:{{ </0> }} {{ translator }}"
```
Comment (zairahira) [inline i18n/locales/urdu/translations.json]:
```suggestion
      "translated-by": "{{ <0> }}مترجم :{{ </0> }} {{ translator }}"
```

File: i18n/locales/urdu/translations.json [hunk 11]
```diff
@@ -32,23 +32,35 @@
...
-  "localization-meta": {
+  "original-author-translator": {
     "roles": {
-      "author": "مصنف",
-      "translator": "مترجم"
+      "author": "مصنف: {{ name }} ({{ locale }})",
+      "translator": "مترجم: {{ name }}"
+    },
+    "details": {
+      "original-article": "{{ <0> }}اصل مضمون:{{ </0> }} {{ title }} از {{ author }}",
```
Comment (zairahira) [inline i18n/locales/urdu/translations.json]:
```suggestion
      "original-article": "{{ <0> }}اصل مضمون:{{ </0> }} {{ title }} مضمون نگار {{ author }}",
```

File: i18n/locales/italian/translations.json (hunk 4 above)
Comment (scissorsneedfoodtoo) [reply i18n/locales/italian/translations.json]: I'll do this one manually if that's alright.

File: i18n/locales/chinese/translations.json (hunk 6 above)
Comment (scissorsneedfoodtoo) [reply i18n/locales/chinese/translations.json]: Hi @miyaliu666, thanks for catching that. And sorry, the mistake here may have been a copy-paste error I made some time ago.

File: i18n/locales/chinese/translations.json (hunk 5 above)
Comment (scissorsneedfoodtoo) [reply i18n/locales/chinese/translations.json]: @miyaliu666, what Ilenia mentioned is right. I simplified this a bit so we don't show the author / translator roles.

Here's what the string looks like in English: `"read-more": "Read {{ <0> }}more posts{{ </0> }}."`

**Review by zairahira — CHANGES_REQUESTED** [review]

File: i18n/locales/arabic/translations.json [hunk 12]
```diff
@@ -32,23 +32,35 @@
     "go-to-front-page": "اذهب للصفحة الامامية"
   },
   "social-row": {
-    "author-has-twitter": "إذا قرأت لهذا الحد ، فغرد للمؤلف لتظهر له أنك تهتم. {{ <0> }}اشكره علي تويتر{{ </0> }}",
-    "author-no-twitter": "إذا كانت هذه المقالة مفيدة, {{ <0> }}شاركها علي تويتر{{ </0> }}.",
-    "default-tweet": "شكرا لك {{ twitter }} لكتابة هذا المقال المفيد."
+    "cta": {
+      "tweet-a-thanks": "إذا قرأت لهذا الحد ، فغرد للمؤلف لتظهر له أنك تهتم. {{ <0> }}اشكره علي تويتر{{ </0> }}",
```
Comment (islamh-mahfouz) [inline i18n/locales/arabic/translations.json]:
```suggestion
      "tweet-a-thanks": "إذا قرأت لهذا الحد ، فغرد للمؤلف لتُظهر له أنك مهتم. {{ <0> }}اشكره علي تويتر{{ </0> }}",
```

File: i18n/locales/arabic/translations.json [hunk 13]
```diff
@@ -32,23 +32,35 @@
...
+      "tweet-it": "إذا كانت هذه المقالة مفيدة, {{ <0> }}شاركها علي تويتر{{ </0> }}."
```
Comment (islamh-mahfouz) [inline i18n/locales/arabic/translations.json]:
```suggestion
      "tweet-it": "إذا كانت هذه المقالة مفيدة، {{ <0> }}شاركها علي تويتر{{ </0> }}."
```

File: i18n/locales/arabic/translations.json [hunk 14]
```diff
@@ -32,23 +32,35 @@
...
+    },
+    "tweets": {
+      "default": "شكرا لك {{ author }} لكتابة هذا المقال المفيد.",
```
Comment (islamh-mahfouz) [inline i18n/locales/arabic/translations.json]:
```suggestion
      "default": "شكرا {{ author }} لكتابة هذه المقالة المفيدة.",
```

File: i18n/locales/arabic/translations.json [hunk 15]
```diff
@@ -32,23 +32,35 @@
...
+      "translation": "شكرًا لك {{ author }} على كتابة هذه المقالة المفيدة ، وشكرًا {{ translator }} على ترجمتها."
```
Comment (islamh-mahfouz) [inline i18n/locales/arabic/translations.json]:
```suggestion
      "translation": "شكرًا {{ author }} لكتابة هذه المقالة المفيدة ، وشكرًا {{ translator }} على الترجمة."
```

File: i18n/locales/arabic/translations.json [hunk 16]
```diff
@@ -32,23 +32,35 @@
...
+  "original-author-translator": {
     "roles": {
-      "author": "المؤلف",
-      "translator": "المترجم"
+      "author": "المؤلف: {{ name }} ({{ locale }})",
+      "translator": "المترجم: {{ name }}"
+    },
+    "details": {
+      "original-article": "{{ <0> }} المقالة الأصلية: {{ </0> }} {{ title }} بقلم {{ author }}",
+      "translated-by": "{{ <0> }} ترجم بواسطة: {{ </0> }} {{ translator }}"
```
Comment (islamh-mahfouz) [inline i18n/locales/arabic/translations.json]:
```suggestion
      "translated-by": "{{ <0> }} تُرجِمت بواسطة: {{ </0> }} {{ translator }}"
```

File: i18n/locales/arabic/translations.json [hunk 17]
```diff
@@ -32,23 +32,35 @@
...
-      "pt-br": "البرتغالية",
-      "it": "الايطالية"
+    "locales": {
+      "english": "الانجليزية",
+      "espanol": "الاسبانية",
+      "chinese": "الصينية",
+      "portuguese": "البرتغالية",
+      "italian": "الايطالية",
+      "japanese": "اليابانية",
+      "arabic": "عربى",
```
Comment (islamh-mahfouz) [inline i18n/locales/arabic/translations.json]:
```suggestion
      "arabic": "العربية",
```

File: i18n/locales/japanese/translations.json [hunk 18]
```diff
@@ -32,23 +32,35 @@
...
-    "author-has-twitter": "お読みいただきありがとうございます。ぜひ著者にツイートしてお知らせください。{{ <0> }}著者宛てにツイート{{ </0> }}",
-    "author-no-twitter": "この記事が参考になりましたらぜひシェアしてください。{{ <0> }}ツイート{{ </0> }}",
-    "default-tweet": "この記事が参考になりました。{{ twitter }} さんありがとうございます。"
+    "cta": {
+      "tweet-a-thanks": "お読みいただきありがとうございます。ぜひ著者にツイートしてお知らせください。{{ <0> }}著者宛てにツイート{{ </0> }}",
+      "tweet-it": "この記事が参考になりましたらぜひシェアしてください。{{ <0> }}ツイート{{ </0> }}"
+    },
+    "tweets": {
+      "default": "この記事が参考になりました。{{ author }} さんありがとうございます。",
+      "translation": "この役立つ記事を書いてくれた{{ author }}と、翻訳してくれた{{ translator }}に感謝します。"
```
Comment (sidemt) [inline i18n/locales/japanese/translations.json]:
```suggestion
      "translation": "この役に立つ記事を書いてくれた {{ author }} さんと、翻訳してくれた {{ translator }} さんに感謝します。"
```

I added spaces around them to make sure the text don't interfere with the twitter handles.

Comment (scissorsneedfoodtoo) [reply i18n/locales/japanese/translations.json]: Awesome, thanks for catching that :slightly_smiling_face:

**Review by majestic-owl448 — DISMISSED** [review]

File: i18n/locales/urdu/translations.json [hunk 19]
```diff
@@ -32,23 +32,35 @@
...
-    "author-has-twitter": "اگر آپ نیں آخر تک پڑھ لیا ہے، تو مصنف کو شکریہ کریں. {{ <0> }} شکریہ کا Tweet کریں {{ </0> }}",
-    "author-no-twitter": " {{ <0> }}Tweet  کریں{{ </0> }} اگر یہ آرٹیکل مفید لگا تو ",
-    "default-tweet": "آپ کا بہت شکریہ، آپ نیں یہ آرٹیکل لکھا.  {{ twitter }} ."
+    "cta": {
+      "tweet-a-thanks": "اگر آپ نیں آخر تک پڑھ لیا ہے، تو مصنف کو شکریہ کریں. {{ <0> }} شکریہ کا Tweet کریں {{ </0> }}",
+      "tweet-it": " {{ <0> }}Tweet  کریں{{ </0> }} اگر یہ آرٹیکل مفید لگا تو "
+    },
+    "tweets": {
+      "default": "آپ کا بہت شکریہ، آپ نیں یہ آرٹیکل لکھا.  {{ author }} .",
+      "translation": "یہ مفید مضمون لکھنے کے لیے {{ author }}، اور {{ translator }} کا ترجمہ کرنے کا شکریہ۔"
```
Comment (zairahira) [inline i18n/locales/urdu/translations.json]:
```suggestion
      "translation": "کا یہ مفید مضمون لکھنے کے لیے  {{ author }}، اور {{ translator }} کا ترجمہ کرنے کا شکریہ۔"
```

**Review by zairahira — CHANGES_REQUESTED** [review]

**Review by majestic-owl448 — DISMISSED** [review]

**Review by zairahira — APPROVED** [review]
Good to go.

File: i18n/locales/urdu/translations.json [hunk 20]
```diff
@@ -32,23 +32,35 @@
...
-    "author-has-twitter": "اگر آپ نیں آخر تک پڑھ لیا ہے، تو مصنف کو شکریہ کریں. {{ <0> }} شکریہ کا Tweet کریں {{ </0> }}",
-    "author-no-twitter": " {{ <0> }}Tweet  کریں{{ </0> }} اگر یہ آرٹیکل مفید لگا تو ",
-    "default-tweet": "آپ کا بہت شکریہ، آپ نیں یہ آرٹیکل لکھا.  {{ twitter }} ."
+    "cta": {
+      "tweet-a-thanks": "اگر آپ نیں آخر تک پڑھ لیا ہے، تو مصنف کو شکریہ کریں. {{ <0> }} شکریہ کا Tweet کریں {{ </0> }}",
+      "tweet-it": " {{ <0> }}Tweet  کریں{{ </0> }} اگر یہ آرٹیکل مفید لگا تو "
+    },
+    "tweets": {
+      "default": "آپ کا بہت شکریہ، آپ نیں یہ آرٹیکل لکھا.  {{ author }} .",
+      "translation": "کا یہ مفید مضمون لکھنے کے لیے  {{ author }}، اور {{ translator }} کا ترجمہ کرنے کا شکریہ۔"
```
Comment (zairahira) [inline i18n/locales/urdu/translations.json]:
```suggestion
      "translation": "مصنف: {{ author }} کا یہ مفید مضمون لکھنے کے لیے، اور مترجم: {{ translator }} کا ترجمہ کرنے کا شکریہ۔"
```

**Review by zairahira — CHANGES_REQUESTED** [review]
It Fixes the RTL alignment issue for thank you tweet messages.

File: i18n/locales/urdu/translations.json (hunk 20 above)
Comment (scissorsneedfoodtoo) [reply i18n/locales/urdu/translations.json]: Awesome, thanks for this fix @zairahira

**Review by sidemt — DISMISSED** [review]
LGTM

**Review by zairahira — APPROVED** [review]

**Review by RafaelDavisH — APPROVED** [review]
LGTM 🚀

**Pull request merged**